erpnext/shopping_cart/cart.py

```diff
@@ -180,6 +180,13 @@ def create_lead_for_item_inquiry(lead, subject, message):
 	lead_doc.update(lead)
 	lead_doc.set('lead_owner', '')
 
+	if not frappe.db.exists('Lead Source', 'Product Inquiry'):
+		frappe.get_doc({
+			'doctype': 'Lead Source',
+			'source_name' : 'Product Inquiry'
+		}).insert(ignore_permissions=True)
+	lead_doc.set('source', 'Product Inquiry')
+
```

Comment:
_**Just a suggestion**_: you can maybe add **Product Enquiry** to `default_lead_sources` in `install_fixtures.py` for new instances.
And also add a small patch to create this Lead Source.
It will be a one time thing and will reduce clutter in this function.
```suggestion
	lead_doc.set('source', 'Product Inquiry')
```

Reply:
@marination what if the user deletes that Lead Source doc?

Reply:
Almost anything can be deleted that way but since this via portal good to be defensive. fair enough @anupamvs